### DISCUSSION TITLE
How to create aliases for virtual users with Dovecot?

### DISCUSSION THREAD

**Discussion by namsral** [stalwartlabs/smtp-server#19] [Q&A]
Coming from postfix, how can I manage aliases for virtual users?

I'm using stalwart-smtp with dovecot-imapd and dovecot-lmtpd.

**Comment by mdecimus** [comment]
If your Dovecot LMTP/IMAP instance has aliases and virtual addresses configured then Stalwart SMTP will automatically handle them.
Before accepting a message, Stalwart SMTP will verify the recipient address against your LMTP server and if the address is accepted then Stalwart SMTP will allow the message to be delivered.

**Comment by namsral** [comment]
To my knowledge it's not the job of the MDA to handle aliases, unless I'm mistaken. In a Postfix/Dovecot setup, it is Postfix which translates an alias to a mailbox.

I'm not tied to Dovecot, but the development of Stalwart's JMAP/IMAP/all-in project are much in flux.

**Reply by mdecimus** [reply]
Stalwart SMTP supports validating addresses from an SQL or LDAP server or alternatively the validation can be delegated to an LMTP or IMAP server.
If your Dovecot is already configured to support aliases then it is easier to delegate the address validation to your Dovecot LMTP/IMAP server.

Aliases and catch-all addresses are configured on the mail storage (in your case Dovecot). SMTP servers such as Stalwart and Postfix could eventually do address rewriting but aliases and virtual addresses are handled by the mail server and stored in a text file, SQL or LDAP directory.

**Reply by namsral** [reply]
It is common for Postfix to handle aliases as Dovecot does not support aliases. I have hit a dead end with my Stalwart-smtp/Dovecot setup, thanks anyway.
